src/pyspiffe/spiffe_id/trust_domain.py

```diff
@@ -1,11 +1,59 @@
+from urllib.parse import urlparse
+
+from pyspiffe import SPIFFE_SCHEME
+
+EMPTY_DOMAIN_ERROR = 'Trust domain cannot be empty'
+SCHEME_SUFFIX = '://'
+
+
 class TrustDomain(object):
     """
     Represents the name of a SPIFFE trust domain (e.g. 'domain.test').
 
-    :param trust_domain: The name of the Trust Domain
+    Args:
+        name(str): The name of the Trust Domain
+
+    Raises:
+        ValueError: is the name of the trust domain is empty, has a port, or contains an invalid scheme.
+
     """
 
-    name: str
+    def __init__(self, name: str):
+        self.__set_name(name)
+
+    def __str__(self):
```

Comment:
Missing typing for method return type"
```
def __str__(self) -> str:
```
I know is obvious here, I just use this one because is the first place where is missing,  We should try to set typing for all the code.